### PR TITLE
Modification of Message struct with generics

### PR DESCRIPTION
**Pull request by saeidalz13**
# Overview

- `Message` struct is now modified to accept generics in order to facilitate the unmarshaling of incoming response messages.
- Attack logic code was commented out in this MR. it was not correct so it will be ready through another MR.
- tests were modified based on the changes

### CODE REVIEW

**Comment by mori-ahk** [comment]
@saeidalz13 is this PR still in progress?

**Comment by saeidalz13** [comment]
This is also good to review yes

File: api/handlers_api.go [hunk 1]
```diff
@@ -43,115 +43,100 @@ func NewWsRequest(server *Server, ws *websocket.Conn, payload ...[]byte) *WsRequ
...
-		game.HostPlayer.IsTurn = true
-	}
-
-	return game, nil
-}
-
-func EndGame(s *Server, ws *websocket.Conn, payload []byte) error {
-	return nil
-}
+// func (w *Request) HandleAttack() (*md.Game, error) {
```
Comment (mori-ahk) [inline api/handlers_api.go:100]: Do we need the commented code?

File: api/utils_api.go [hunk 2]
```diff
@@ -24,20 +23,20 @@ func SendMsgToBothPlayers(game *md.Game, hostMsg, joinMsg *md.Message) error {
 	return nil
 }
 
-func ExtractFindGamePlayer(server *Server, initMap map[string]interface{}) (*md.Game, *md.Player, error) {
-	desiredStrings, err := TypeAssertStringPayload(initMap, md.KeyGameUuid, md.KeyPlayerUuid)
-	if err != nil {
-		return nil, nil, err
-	}
-	gameUuid, playerUuid := desiredStrings[0], desiredStrings[1]
+// func ExtractFindGamePlayer(server *Server, initMap map[string]interface{}) (*md.Game, *md.Player, error) {
```
Comment (mori-ahk) [inline api/utils_api.go:26]: same here

File: api/utils_api.go [hunk 3]
```diff
@@ -24,20 +23,20 @@ func SendMsgToBothPlayers(game *md.Game, hostMsg, joinMsg *md.Message) error {
...
-	if game == nil {
-		return nil, nil, cerr.ErrGameNotExists(gameUuid)
-	}
-	player := server.FindPlayer(playerUuid)
-	if player == nil {
-		return nil, nil, cerr.ErrPlayerNotExist(playerUuid)
-	}
-	return game, player, nil
-}
+// 	game := server.FindGame(gameUuid)
```
Comment (mori-ahk) [inline api/utils_api.go:33]: same here

File: api/ws_api.go [hunk 4]
```diff
@@ -204,48 +201,47 @@ func (s *Server) manageWsConn(ws *websocket.Conn) {
...
-					md.WithPayload(md.RespAttack{
-						IsTurn: game.JoinPlayer.IsTurn},
-					),
-				)
-				if err := SendMsgToBothPlayers(game, &hostMsg, &joinMsg); err != nil {
-					log.Println(err)
-				}
-				continue
-			}
+		// 		if err := ws.WriteJSON(respErr); err != nil {
```
Comment (mori-ahk) [inline api/ws_api.go:219]: same here

**Review by mori-ahk — COMMENTED** [review]
Left some comments about commented code. Once those are taken care of we can merge.

**Comment by saeidalz13** [comment]
I have taken care of all these issues and also modified the structures in `simplify_codes`. we can ignore this branch and just merge that one. it will have all the desired changes

**Pull request closed without merging**